ash_dal/dao/async_dao.py

```diff
@@ -57,3 +57,23 @@ async def paginate(self, page_size: int | None = None) -> t.AsyncIterator[Pagina
             async for page in paginator.paginate():
                 entities = self._get_entities_from_db_items(db_items=page)
                 yield PaginatorPage(index=page.index, items=entities)
+
+    async def filter(self, specification: dict[str, t.Any]) -> tuple[Entity, ...]:
+        async with self.db.session as session:
+            db_items = await session.scalars(select(self.__model__).filter_by(**specification))
+            return self._get_entities_from_db_items(db_items=db_items)
+
+    async def filter_paginated(
```

Comment:
this method is practically the same as just paginate. Maybe it make sense to left just paginate and make it possible to accept filter. So specification will be optional and will be applied if you provide it(if it is not None).